src/datasources.js

```diff
@@ -225,11 +221,59 @@ define([
         return widget;
     }
 
+    function fixtureWidget(mug, options, labelText) {
```

Comment:
Yeah. I was going to do something like that when I generalized everything to beyond fixtures. I thought may be good to keep it as fixtureWidget until then